legacy_packages/sdk/src/evm/common/sign.ts

```diff
@@ -54,10 +54,17 @@ export async function signTypedDataInternal(
 
   // an indirect way for accessing walletconnect's underlying provider
   if ((provider as any)?.provider?.isWalletConnect) {
-    signature = await provider.send("eth_signTypedData", [
-      signerAddress,
-      JSON.stringify(payload),
-    ]);
+    try {
+      signature = await provider.send("eth_signTypedData", [
```

Comment:
one thing that's annoying is if the user denies the sig, they have to deny twice since we try again on catch

Reply:
To avoid this I'll prioritize the v4 version and if that throws with the unsupported method error I'll try with signTypedData basic otherwise throw